docs/git/branches.md

```diff
@@ -0,0 +1,169 @@
+
+Or we could only make the ice cream change, then send it for review, then when that merges, write the story and send it for review. But that slows us down, and also means that the reviewer can't review both changes at the same time.
+
+It would be better if we could send out one pull request for the ice cream change, and another for the story. We can do this - with branches:
+
+TODO: Same image as before, of divergent branches, but labelled "ice cream" and "story".
+
+We can then add new commits on top of each branch, separately, and later on merge both branches to main.
+
+### How do we create a new branch?
+
+```console
+$ git switch --create story
+Switched to a new branch 'story'
+```
+
+Let's run through the bits of that command:
+
+`git` - we're running a git command.
+
+`switch` - we're switching to a different branch than the one we're on right now.
+
+`--create` - we want git to create the branch we're switching to.
+
+`story` - the branch we want to switch to (and create) is called `story`.
+
+Running this command doesn't change any of the files in our repository - they stay the same, but if we commit changes, they will only get committed on this branch. If we switched back to the `main` branch, the changes would be un-done. But if we switch back to `story`, they'll come back!
+
+### How do I know what branch I'm on?
+
+You've already run one command which tells you this, you just may not have noticed!
+
+```console
+$ git status
+On branch story
+nothing to commit, working tree clean
+```
+
+That's right, `git status` tells you what branch you're on (and about any changes you haven't committed).
+
+You can also find out more about branches by running:
+
+```console
+$ git branch
+  main
+* story
+```
+
+`git branch` tells you what branches exist in your repository, and puts a star next to the one you're currently switched to.
+
+### Exercise 1 (15 minutes)
+
+Let's make a new branch, and make a new commit on it.
+
+Create a branch named story, and edit `mug-brownie.md`.
+
+Before the "What you need" heading, make up a story about how the author discovered the recipe, or a time they enjoyed it.
+
+When you've written a bit, make a commit (make sure to give it a good commit message!)
+
+Explore the history with `git log` - make sure you understand the commits in your history and why they're there.
+
+### Switching back to main
+
+Can you guess how to switch back to the `main` branch?
+
+What do you think will happen when you do?
+
+What will the history look like?
+
+### Exercise 2 (5 minutes)
+
+Switch back to the `main` branch (hint: The command to run is like the one you used to create a new branch, but you're not creating a new one).
+
+Look at the contents of `mug-brownie.md` - what's changed?
+
+Explore the history with `git log` - make sure you understand the commits in your history and why they're there.
+
+Share with the class what you saw.
```

Comment:
In the past I've found stuff like this a bit unstructured. Discussing with a buddy/small group (and maybe reporting back to the larger group) works better.